testsuite/model/src/test/java/org/keycloak/testsuite/model/parameters/HotRodMapStorage.java

```diff
@@ -100,18 +109,16 @@ public void updateConfig(Config cf) {
                 .config("keyType.single-use-objects", "string");
 
         cf.spi(HotRodConnectionSpi.NAME).provider(DefaultHotRodConnectionProviderFactory.PROVIDER_ID)
-                .config("enableSecurity", "false")
-                .config("configureRemoteCaches", "false");
+                .config("host", hotRodContainer.getContainerIpAddress())
+                .config("port", String.valueOf(hotRodContainer.getMappedPort(11222)))
+                .config("username", USERNAME)
+                .config("password", PASSWORD)
+                .config("configureRemoteCaches", "true");
     }
 
     @Override
     public void beforeSuite(Config cf) {
-        hotRodServerRule.createHotRodMapStoreServer();
-    }
-
-    @Override
-    public Statement classRule(Statement base, Description description) {
-        return hotRodServerRule.apply(base, description);
+        hotRodContainer.start();
```

Comment:
How is the container stopped?

Reply:
This is a good question. I was trying to make it stop, but I was not able to hook `afterSuite`. Then I realized the container is stopped after the tests (`docker ps` doesn't show anything), so probably testcontainers are doing it somehow.

Reply:
I added `afterSuite` support using junit `RunListener`. It works correctly with maven.

Sadly, it doesn't work using Intellij Junit runner, it seems the listener is not registered at all by Intellij. On the other hand, as I said in the previous comment, the container is stopped automatically, but it takes a few seconds after the run is finished.